### ISSUE TITLE
Error when upgrading golang packages

### ISSUE THREAD

**Issue by ikbear** [r-darwish/topgrade#376]
**What did you expect to happen?**

Upgrade all golang packages.

**What actually happened?**

`go get all: cannot match "all": working directory is not part of a module`

**Additional details:**
- Which operating system or Linux distribution are you using?
  - Mac OS X 10.15.3 (19D76)
- How did you install Topgrade?
  - Download from old release and then upgrade to v4.2.0.
- If a specific step is failing, what happens when you run the command on your own? (Run `topgrade --dry-run` to see which commands Topgrade is running)
`Dry running: /usr/local/bin/go get -u all`
<details><pre>
➜ topgrade -v
 2020-03-17T08:27:42.465Z DEBUG topgrade::config > Configuration at /Users/ikbear/.config/topgrade.toml
 2020-03-17T08:27:42.465Z DEBUG topgrade::config > Loaded configuration: ConfigFile { pre_commands: None, commands: None, git_repos: None, predefined_git_repos: None, disable: None, remote_topgrades: None, ssh_arguments: None, git_arguments: None, tmux_arguments: None, set_title: None, assume_yes: None, yay_arguments: None, no_retry: None, run_in_tmux: None, cleanup: None, notify_each_step: None, accept_all_windows_updates: None, only: None }
 2020-03-17T08:27:42.466Z DEBUG topgrade         > Version: 4.2.0
 2020-03-17T08:27:42.466Z DEBUG topgrade         > OS: x86_64-apple-darwin
 2020-03-17T08:27:42.466Z DEBUG topgrade         > Args { inner: ["topgrade", "-v"] }
 2020-03-17T08:27:42.466Z DEBUG topgrade         > Binary path: Ok("/usr/local/bin/topgrade")
 2020-03-17T08:27:42.466Z DEBUG topgrade         > Self Update: false
 2020-03-17T08:27:42.466Z DEBUG topgrade::utils  > Detected "/usr/bin/git" as "git"
 2020-03-17T08:27:42.466Z DEBUG topgrade::utils  > Detected "/usr/bin/sudo" as "sudo"
 2020-03-17T08:27:42.467Z DEBUG topgrade::utils  > Cannot find "pwsh"
 2020-03-17T08:27:42.467Z DEBUG topgrade::utils  > Cannot find "powershell"
 2020-03-17T08:27:42.467Z DEBUG topgrade::runner > Step "brew"
 2020-03-17T08:27:42.467Z DEBUG topgrade::utils  > Detected "/usr/local/bin/brew" as "brew"
</pre></details>

**Comment by andrei-pavel** [comment]
It seems like it's taking `all` as a literal Go module rather than a reserved keyword for retrieving all modules.

Could you show the output of `go version`? It's working for me on `go version go1.14 linux/amd64`.

**Comment by ikbear** [comment]
`➜ go version
go version go1.14 darwin/amd64`

The latest version for Mac.

**Comment by andrei-pavel** [comment]
Documentation is rather well hidden: https://golang.org/cmd/go/#hdr-Package_lists_and_patterns

```
- "all" expands to all packages found in all the GOPATH trees. For example, 'go list all' lists all the packages on the local system. When using modules, "all" expands to all packages in the main module and their dependencies, including dependencies needed by tests of any of those.
```

**Comment by andrei-pavel** [comment]
Digressing disclaimer is I've been running `topgrade` with `--disable go` for a while because of another issue with it. Which is that I get a boatload of `cannot find package` and one scenario where I've pinned this down is:
Repo A has B as dependency.
Repo A has B version-locked at V.
`go get -u all` updates B to the latest version > V.
B at version V no longer has required module by A.
Hence the error.

So my assumption is, either you have repos which keep their dependencies up to date regularly or which don't version-lock them, or this command doesn't work.

So if this `cannot match "all"` adds to the list of problems and if we can't get to the bottom of these, then I suggest that go would be disabled from running by default, but maybe keep the code for later enablement.

**Comment by ikbear** [comment]
> So my assumption is, either you have repos which keep their dependencies up to date regularly or which don't version-lock them, or this command doesn't work.

I also think so. Thanks for your sharing.

**Comment by andrei-pavel** [comment]
Right, but that applies in my case. I'm trying to replicate your issue for now. Could you show output of `go env` run somewhere not in a Go project, maybe in your home directory?

**Comment by r-darwish** [comment]
This should be fixed in e6546710c675d27b529caa8778e1e906ccf1f615. If this issue still happens in the next version of Topgrade please reopen this issue.

**Comment by ikbear** [comment]
> Right, but that applies in my case. I'm trying to replicate your issue for now. Could you show output of `go env` run somewhere not in a Go project, maybe in your home directory?

Maybe I have something been setting wrong in go1.3. I will figure out it and reply to you if any new update.
